### ISSUE TITLE
Recognise "method" keyword as a subroutine

### ISSUE THREAD

**Issue by davorg** [matisse/Perl-Metrics-Simple#18]
I've been using Perl::Metrics::Simple to automatically measure the complexity of a lot of my CPAN code using GitHub Actions (see [workflow file here](https://github.com/PerlToolsTeam/github_workflows/blob/main/.github/workflows/cpan-complexity.yml).

Today, I looked at one of the [reports for App::MergeCal](https://github.com/davorg-cpan/app-mergecal/actions/runs/9828670523/job/27132873813) (see the "run countperl" job). I spotted that the code had a rather high complexity because it's all counted as "code not in named subroutines". I assume that's because I'm using the new [perlclass](https://perldoc.perl.org/perlclass) feature and most of my code is in methods, not subroutines.

So this is just here to alert you to the issue. My plan is to look at your code and see if I can work out how to fix this and send you a pull request over the next couple of days.

**Comment by matisse** [comment]
Hi Dave and thanks for the report.

It is indeed true that `Perl::Metrics::Simple` relies on the `sub` keyword to find subroutines.
See [Perl/Metrics/Simple/Analysis/File.pm line 141](https://metacpan.org/release/MATISSE/Perl-Metrics-Simple-v1.0.3/source/lib/Perl/Metrics/Simple/Analysis/File.pm#L141)

 It does this by with the [[PPI](https://metacpan.org/pod/PPI::Statement::Sub)](https://metacpan.org/pod/PPI::Statement::Sub)

I believe that currently there is not a PPI::Statement subclass for `method`

So, I am curious what solution you think might be helpful here?

**Comment by davorg** [comment]
Yeah, I think we need to start by adding a `method` class to PPI. I'll look into that.

**Comment by davorg** [comment]
See https://github.com/Perl-Critic/PPI/issues/299

**Comment by matisse** [comment]
Noting for the record here the PPI  
[RFC: proof-of-concept for feature-tracking and perl sub signatures](https://github.com/Perl-Critic/PPI/pull/280)
